rustworkx-core/src/connectivity/cycle_basis.rs

```diff
@@ -117,9 +117,160 @@ where
     cycles
 }
 
+/// Return a list of edges representing cycles which form a basis for cycles of a given graph.
+///
+/// A basis for cycles of a graph is a minimal collection of
+/// cycles such that any cycle in the graph can be written
+/// as a sum of cycles in the basis.  Here summation of cycles
+/// is defined as the exclusive-or of the edges.
+///
+/// This is adapted from
+///    Paton, K. An algorithm for finding a fundamental set of
+///    cycles of a graph. Comm. ACM 12, 9 (Sept 1969), 514-518.
+///
+/// The function implicitly assumes that there are no parallel edges.
+/// It may produce incorrect/unexpected results if the input graph has
+/// parallel edges.
+///
+///
+/// Arguments:
+///
+/// * `graph` - The graph in which to find the basis.
+/// * `root` - Optional node index for starting the basis search. If not
+///     specified, an arbitrary node is chosen.
+///
+/// # Example
+/// ```rust
+/// use petgraph::prelude::*;
+/// use rustworkx_core::connectivity::cycle_basis_edges;
+///
+/// let edge_list = [(0, 1), (0, 3), (0, 5), (1, 2), (2, 3), (3, 4), (4, 5)];
+/// let graph = UnGraph::<i32, i32>::from_edges(&edge_list);
+/// let mut res: Vec<Vec<(NodeIndex, NodeIndex)>> = cycle_basis_edges(&graph, Some(NodeIndex::new(0)));
+/// ```
+pub fn cycle_basis_edges<G>(graph: G, root: Option<G::NodeId>) -> Vec<Vec<(G::NodeId, G::NodeId)>>
```

Comment:
Instead of returning `Vec<(G::NodeId, G::NodeId)>` here it might be better to return `Vec<G::EdgeId>`. Then if we want to return the edge endpoint or weight we can just look it up from the index.

Reply:
It's funny you mention this, as in an earlier version, I had the function return the `EdgeId` attribute of each `EdgeRef` but replaced it with the tuple thinking that it should instead return the edge objects. 

Also, I'm assuming that by returning the edge endpoint you mean returning it using the wrapper located in `rustworkx/src/connectivity/mod.rs` lines 100 to 110. Should I use the graph.edges() method fetch the edges by index?

Reply:
By edge endpoints I meant what you're doing now. For each edge you find you're doing `(edge.source(), edge.target())` (the endpoints being the tuple of nodes for the edge), but instead if you did `edge.id()` that would give you the unique `G::EdgeId` for the edge. Then if you returned a `Vec<G::EdgeId>` callers of this function could look up any attributes of the edge they needed from that list. For examples if you wanted a list of endpoint tuples you could do something like:
```rust
let edge_endpoints = cycle_basis(graph, None)
    .into_iter()
    .filter_map(|e| graph.edge_endpoints(e))
    .collect()
```

but it would also give you the flexibility to get the weights, or other details. It basically is just returning a reference to the edge in `graph` instead of one of properties of the edge.

Reply:
I see, I will make those changes accordingly.